### ISSUE TITLE
Low subtitle resolution (maybe due to video resolution)

### ISSUE THREAD

**Issue by SonicPotions** [jellyfin/jellyfin-web#2248]
**Describe The Bug**
I noticed on some shows the subtitles have a low resolution / look pixelated.
I'm not sure if it is due to the source material being 720 or 1080p.
On One show I watch (1080p) they are crisp. 
On a 720p show I watch they are pixelated and have rough edges. as if no anti aliasing happens.

**Steps To Reproduce**
activate subtitles (on 720p? file)
see that they are pixelated / have rough edges

**Expected Behavior**
crisp subtitles

**Screenshots**
Crisp subtitle on 1080p media
![crisp1080pSubtitles](https://user-images.githubusercontent.com/3105470/102712889-91149300-42c4-11eb-9cbd-66c90c72bfa9.png)

rough subtitles on 720p media
![pixelated720pSubtitles](https://user-images.githubusercontent.com/3105470/102712898-a1c50900-42c4-11eb-8aa6-db7cd28a234c.png)

**System (please complete the following information):**
 - Linux web app and Fire TV 4k app
 - Firefox
 - Jellyfin Version: 10.6.4

**Comment by nyanmisaka** [comment]
What's the type of this subtitle? SRT, ASS/SSA or PGSSUB?
Are subtitles being burned into the video?

**Comment by SonicPotions** [comment]
the 720p video with the pixelated subtitles has the following info:

Container mkv
Path/tv/Konosuba/Season 1/KonoSuba.S01E01.German.AC3.DL.720p.BluRay.x264-Fuuko.mkv
Size849 MB

**Video**
Title720p H264
CodecH264
AVCNo
ProfileHigh
Level41
Resolution1280x720
Aspect ratio16:9
InterlacedNo
Framerate23.976025
Bitrate4712 kbps
Bit depth8 bit
Pixel formatyuv420p
Ref frames1
NAL0

**Subtitle**
TitleGer
Languageger
CodecPGSSUB
DefaultNo
ForcedNo
ExternalNo

How to find out if they are burned in?
For the crisp ones I know since the Fire TV app displays a message box "burned in subtitles may take a while" and a loading screen is shown. With the 720p version the subtitles showed up more or less immediately when I select them.

**Comment by SonicPotions** [comment]
at least in the web player I can display some info:

**Playback Info**
Player:
Html Video Player
Play method:
Transcode
Protocol:
http
Stream type:
HLS

**Video Info**
Player dimensions:
1271x900
Video resolution:
1280x720
Dropped frames:
0
Corrupted frames:
0

**Transcoding Info**
Video codec:
H264
Audio codec:
MP3
Bitrate:
5.1 Mbps
Transcoding progress:
98.1%
Transcoding framerate:
201 fps
Reason for transcoding:
SubtitleCodecNotSupported

**Original Media Info**
Container:
webm
Size:
848.7 MiB
Bitrate:
5.5 Mbps
Video codec:
H264 High
Video bitrate:
4.7 Mbps
Audio codec:
AC3
Audio bitrate:
384 kbps
Audio channels:
2
Audio sample rate:
48000 Hz

**Comment by SonicPotions** [comment]
the 1080p video with the crisp subtitles also has PGSSUBs

**Playback Info**
Player:
Html Video Player
Play method:
Transcode
Protocol:
http
Stream type:
HLS

**Video Info**
Player dimensions:
1271x900
Video resolution:
1920x1080
Dropped frames:
0
Corrupted frames:
0

**Transcoding Info**
Video codec:
H264
Audio codec:
MP3
Bitrate:
12.4 Mbps
Transcoding progress:
82.8%
Transcoding framerate:
92 fps
Reason for transcoding:
SubtitleCodecNotSupported

**Original Media Info**
Container:
webm
Size:
2 GiB
Bitrate:
13.3 Mbps
Video codec:
H264 High
Video bitrate:
12.0 Mbps
Audio codec:
AC3
Audio bitrate:
640 kbps
Audio channels:
2
Audio sample rate:
48000 Hz

**Comment by nyanmisaka** [comment]
Can I have the ffmpeg transcoding log?

**Comment by SonicPotions** [comment]
here is the log. I just started the show for a few seconds and displayed the choppy subtitles to generate the log

[KonosubaLog.txt](https://github.com/jellyfin/jellyfin-web/files/5723557/KonosubaLog.txt)

**Comment by SonicPotions** [comment]
I extracted a subtitle image from both videos. quality looks equal on both.
These are direct extracted images of the PGS subs

from the pixelated 720p file
![720pSub](https://user-images.githubusercontent.com/3105470/102766147-faa4a800-437d-11eb-8aec-9ad995a91e54.png)

from the crisp 1080p file
![1080pSub](https://user-images.githubusercontent.com/3105470/102766154-fd070200-437d-11eb-96cb-757e89b06235.png)

**Comment by nyanmisaka** [comment]
`-filter_complex "[0:3]scale=1280x720[sub];[0:0]scale=trunc(min(max(iw\,ih*dar)\,1920)/2)*2:trunc(ow/dar/2)*2[base];[base][sub]overlay"`

The graphical subtitle is being burned into the video.

**Comment by nyanmisaka** [comment]
```
/usr/lib/jellyfin-ffmpeg/ffmpeg -f matroska,webm -i file:"/tv/Konosuba/Season 1/KonoSuba.S01E01.German.AC3.DL.720p.BluRay.x264-Fuuko.mkv" -map_metadata -1 -map_chapters -1 -threads 0 -map 0:0 -map 0:2 -codec:v:0 libx264 -pix_fmt yuv420p -preset veryfast -crf 23 -maxrate 4712307 -bufsize 9424614 -profile:v high -level 4.1 -x264opts:0 subme=0:me_range=4:rc_lookahead=10:me=dia:no_chroma_me:8x8dct=0:partitions=none  -force_key_frames:0 "expr:gte(t,687+n_forced*3)" -g 72 -keyint_min 72 -sc_threshold 0 -filter_complex "[0:3]scale=1280x720[sub];[0:0]scale=trunc(min(max(iw\,ih*dar)\,1920)/2)*2:trunc(ow/dar/2)*2[base];[base][sub]overlay" -an -sn -y /tmp/out.mp4
```

Can you try to change to algorithm in scale filter to find which is the best one for your subtitle?
`scale=s=1280x720:flags=bilinear`

```
-sws_flags         <flags>      E..V...... scaler flags (default bicubic)
     fast_bilinear                E..V...... fast bilinear
     bilinear                     E..V...... bilinear
     bicubic                      E..V...... bicubic
     experimental                 E..V...... experimental
     neighbor                     E..V...... nearest neighbor
     area                         E..V...... averaging area
     bicublin                     E..V...... luma bicubic, chroma bilinear
     gauss                        E..V...... Gaussian
     sinc                         E..V...... sinc
     lanczos                      E..V...... Lanczos
     spline                       E..V...... natural bicubic spline
```

**Comment by SonicPotions** [comment]
I'll give it a try, probably tonight, gotta make some cookies with the kids now.

**Comment by SonicPotions** [comment]
nope... different scaling filters result just in different kinds of blurry. either it is sharp and pixelated or soft and blurry. Not sure what I'd prefer. Maybe I'll post some example images to compare the results later.

But to check if maybe just the PGS is to blame, the subtitles look as ugly in VLC on my desktop as they do in the jellyfin players. So I guess its not a bug, but just a low quality media file and my missing knowledge of bitmap based subtitles.

**Comment by SonicPotions** [comment]
okay, I did some more testing and it really seems to be the medias fault. I even tried setting up a filter chain to oversample the subs, but they did not really get better using only ffmpeg. So I'm gonna close this bug.
it seems others have those pixelated subs as well (vlc or some guys in DVD authoring forums) and on later episodes I saw the subs are clearer even though the material is still 720p.

sorry for making all that fuzz, as it turns out I did just not know enough about PGS subs, since I guess all subs I used before where .srt
